src/utils/schema-utils.ts

```diff
@@ -39,14 +39,20 @@ const filterInferSchemaResponse = (schema: any) => {
                 return hasLength(inferredProperty.pointer);
             })
             .map((inferredProperty: any) => {
-                const interrefPropertyTypes = inferredProperty.types;
+                const inferredPropertyTypes: string[] = inferredProperty.types;
                 const isValidKey = Boolean(
                     // Happens when the schema contradicts itself, which isnt a "feature" we use intentionally
                     inferredProperty.exists !== 'cannot' &&
```

Comment:
I'm guessing this check is still fine the way it is, but I wanted to mention this because I realized I may have forgot to tell you about something important 😁 

Technically, there's an [`Exists::Implicit` variant](https://github.com/estuary/flow/blob/db2cdd86825132ee7e0bcac8b432712ab5866c83/crates/doc/src/inference.rs#L1132), which would be returned if you asked schema inference about a property that isn't explicitly mentioned in the schema, but is allowed to exist due to something like `additionalProperties`. Not expecting that to be relevant here, since you're only using the pointers you got back from inference. But mentioning it just in case.